benchexec/result.py

```diff
@@ -314,16 +332,93 @@ def get_result_classification(result):
         return RESULT_CLASS_FALSE
 
 
-def get_result_category(filename, result, properties):
+def _compare_multiproperty_statuses(actual_statuses, ideal_statuses):
+    assert not (bool(ideal_statuses.keys() - actual_statuses.keys()))
+    is_correct = True
+    is_error = False
+    is_unknown = False
+    for property in ideal_statuses.keys():
+        actual_status = str(actual_statuses[property]).lower()
+        ideal_status = str(ideal_statuses[property]).lower()
+        if actual_status == RESULT_CLASS_TRUE and ideal_status == RESULT_CLASS_FALSE or \
+                                actual_status == RESULT_CLASS_FALSE and ideal_status == RESULT_CLASS_TRUE:
+            is_correct = False
+        if actual_status == RESULT_CLASS_ERROR and \
+                (ideal_status == RESULT_CLASS_FALSE or ideal_status == RESULT_CLASS_TRUE):
+            is_error = True
+        if actual_status == RESULT_CLASS_UNKNOWN and \
+                (ideal_status == RESULT_CLASS_FALSE or ideal_status == RESULT_CLASS_TRUE):
+            is_unknown = True
+    if not is_correct:
+        return CATEGORY_WRONG
+    if is_error:
+        return CATEGORY_ERROR
+    if is_unknown:
+        return CATEGORY_UNKNOWN
+    return CATEGORY_CORRECT
+
+
+def _score_for_multiproperty(actual_statuses, ideal_statuses, result):
+    assert not (bool(ideal_statuses.keys() - actual_statuses.keys()))
+    score = 0
+    for property in ideal_statuses.keys():
+        actual_status = str(actual_statuses[property]).lower()
+        ideal_status = str(ideal_statuses[property]).lower()
+
+        if ideal_status == RESULT_CLASS_TRUE:
+            if not result:
+                score += _SCORE_CORRECT_TRUE
+            elif actual_status == RESULT_CLASS_TRUE:
+                score += _SCORE_CORRECT_TRUE
+            elif actual_status == RESULT_CLASS_FALSE:
+                score += _SCORE_WRONG_FALSE
+            else:
+                score += _SCORE_UNKNOWN
+        elif ideal_status == RESULT_CLASS_FALSE:
+            if not result:
+                score += _SCORE_CORRECT_FALSE
+            elif actual_status == RESULT_CLASS_TRUE:
+                score += _SCORE_WRONG_TRUE
+            elif actual_status == RESULT_CLASS_FALSE:
+                score += _SCORE_CORRECT_FALSE
+            else:
+                score += _SCORE_UNKNOWN
+        else:
+            score += _SCORE_UNKNOWN
+    return score
+
+
+def _get_yaml_ideal_statuses(filename):
+    # Multiple properties are being checked.
+    # We need to get results from specific YAML file.
+    ideal_statuses = {}
+    specification_file = filename + _YAML_EXTENSION
+    with open(specification_file, 'r') as f:
```

Comment:
This just crashes if the file is not present or cannot be read. Proper error handling should be added.